usagov/vote-gov-drupal
# Vote-1483 -> Reorganize Testing Folder

**_Delete all details that do not apply to this PR!_**

## Jira ticket (required)
[Vote-1483](https://bixal-projects.atlassian.net/browse/VOTE-1483)

## Description (optional)
This PR will reorganize our testing folder and break up tests into frontend and backend tests 

All tests will be able to run locally 

- front end will hit the static site in the pipeline 
- back end tests will only be run locally for the time being 

> **Note**
> I have included the `npm run cy:pipeline` command and the new `pipeline` config file to be merged into dev.. how it is set up is testing will only be run on stage, dev, and test.  The work has been tested on the `test ` branch and screenshots have been added in the files changed to show that it is running

## Deployment and testing (required, if applicable)
### Testing steps
1. pull down the branch
2. start local 
3. cd into `testing`
4. run the following commands `npm run cy:backEnd` and `npm run cy:frontEnd` this will run both the backend and frontend headless 
5. run `npm run cy:createUser` and then run `npm run cy:open` this will open the runner and click through both backend and frontend tests and ensure test are passing.  Once the tests are running return back to the terminal and run `npm run cy:delteUser`
